src/main/java/de/dhbw/mh/slang/craft/parser/CraftedSlangParser.java

```diff
@@ -96,26 +103,44 @@ public AstNode conditionalExpression( ){
 	
 	@Override
 	public AstNode conditionalOrExpression( ){
-		// TODO Auto-generated method stub
-		return super.conditionalOrExpression( );
+		switch(LEXER.lookahead().TYPE) {
+			case PLUS:
+			case MINUS:
+			case LPAREN:
+			case IDENTIFIER:
+			case NUMERIC_LITERAL:
+				return this.disjunction(conditionalAndExpression());
+			default:
+				Set<Token.Type> acceptedTypes = setOf(PLUS, MINUS, LPAREN, IDENTIFIER, NUMERIC_LITERAL);
+				throw this.parsingException(acceptedTypes);
+		}
 	}
 	
 	@Override
 	AstNode disjunction( AstNode previous ){
-		// TODO Auto-generated method stub
-		return super.disjunction( previous );
+		switch(LEXER.lookahead().TYPE) {
+			case LOR:
+				return this.disjunction1(previous);
+			case EOF:
+			case RPAREN:
+				return this.disjunction2(previous);
+			default:
+				throw this.parsingException(Selector.DISJUNCTION);
+		}
 	}
 	
 	@Override
 	AstNode disjunction1( AstNode previous ){
-		// TODO Auto-generated method stub
-		return super.disjunction1( previous );
+		if(LEXER.lookahead().TYPE != LOR) {
+			throw this.parsingException(LOR);
+		}
+
+		return new AstBinaryOperation(this.LEXER.lookahead().BEGIN,previous, Operator.LOGICAL_OR ,conditionalAndExpression());
```

Comment:
The JavaDoc specifies the grammar rule `disjunction : '||' conditionalAndExpression disjunction ;` for this method. But I can see only one call of `conditionalAndExpression`. Could it be that `disjunction` satisfies the remaining tests? :wink:

Reply:
well in that case we'd need to add a call to disjunction somewhere.. We figured it may be possible to do this similar to how we did it in conditionalOrExpression (passing conditionalAndExpression as parameter to disjunction) however that doesn't get us anywhere either...
Not quite sure how to proceed

Reply:
I just know that we forgot to add a LEXER.advance()

Reply:
fixed this one with [d603317](https://github.com/university-lectures/slang/pull/12/commits/d603317a1e056513b65700c2c45ec7a36332e4cc)
One test still failing tho